app/models/concerns/patient_searchable.rb

```diff
@@ -7,62 +7,20 @@ module PatientSearchable
   class_methods do
     # Case insensitive and phone number format agnostic!
     def search(name_or_phone_str, lines = LINES)
-      # lines should be an array of symbols
-      name_regexp = /#{Regexp.escape(name_or_phone_str)}/i
+      wildcard_name = "%#{name_or_phone_str}%"
       clean_phone = name_or_phone_str.gsub(/\D/, '')
-      phone_regexp = /#{Regexp.escape(clean_phone)}/
-      identifier_regexp = /#{Regexp.escape(name_or_phone_str)}/i
 
-      all_matching_names = find_name_matches name_regexp, lines
-      all_matching_phones = find_phone_matches phone_regexp, lines
-      all_matching_identifiers = find_identifier_matches(identifier_regexp)
-
-      sort_and_limit_patient_matches(all_matching_names, all_matching_phones, all_matching_identifiers)
-    end
-
-    private
-
-    def sort_and_limit_patient_matches(*matches)
-      all_matches = matches.reduce do |results, matches_of_type|
-        results | matches_of_type
-      end
-
-      all_matches.sort { |a, b|
-        b.updated_at <=> a.updated_at
-      }.first(SEARCH_LIMIT)
-    end
-
-    def find_name_matches(name_regexp, lines = LINES)
-      if nonempty_regexp? name_regexp
-        primary_names = Patient.in(line: lines).where name: name_regexp
-        other_names = Patient.in(line: lines).where other_contact: name_regexp
-        return (primary_names | other_names)
+      base = Patient.where(line: lines)
+      matches = base.where('name ilike ?', wildcard_name)
+                    .or(base.where('other_contact ilike ?', wildcard_name))
+                    .or(base.where('identifier ilike ?'), wildcard_name)
+      if clean_phone.present?
+        clean_phone = "%#{clean_phone}%"
+        matches = matches.or(base.where('primary_phone like ?', clean_phone))
+                         .or(base.where('other_phone like ?', clean_phone))
```

Comment:
Does this scope properly? I think it does but since the test is currently failing from the subobjects and I get paid to think in JS now I feel a bit uneasy

Reply:
It does. The changes should be identical to https://github.com/DCAFEngineering/dcaf_case_management/pull/2123, and we explicitly test for this in https://github.com/DCAFEngineering/dcaf_case_management/blob/main/test/models/patient/searchable_test.rb#L69 .

(This did reveal that one of the parens was in the wrong spot, fixed in a commit I just added.)

You can try it yourself by opening up the Patient model, commenting out `after_create :initialize_fulfillment` and `validates_associated :fulfillment`, then pasting this into a `rails c` (empty db is fine):

```
Patient.create name: 'Cat', primary_phone: '1234567777', line: 'DC', initial_call_date: Time.zone.now
Patient.create name: 'Cat', primary_phone: '1234567775', line: 'MD', initial_call_date: Time.zone.now
Patient.search('cat').count # 2
Patient.search('cat', 'DC') # 1
````

Reply:
or, maybe you mean phone?

```
Patient.create name: 'Cat', primary_phone: '1234567777', line: 'DC', initial_call_date: Time.zone.now
Patient.create name: 'Cat', primary_phone: '1234567775', line: 'MD', initial_call_date: Time.zone.now, other_phone: '1234567777'
Patient.search('1234567777').count # 2
Patient.search('1234567777', 'DC') # 1
```

Reply:
ty for entertaining my paranoia!